src/psyclone/tests/domain/lfric/arg_index_to_metadata_index_test.py

```diff
@@ -0,0 +1,175 @@
+        :py:class:`psyclone.domain.lfric.kernel.LFRicKernelMetadata`
+
+    :returns: an ArgIndexToMetadataIndex class
+    :rtype: :py:class:`psyclone.domain.lfric.ArgIndexToMetadataIndex`
+
+    '''
+    metadata.validate()
+    cls = ArgIndexToMetadataIndex
+    cls._initialise(None)
+    cls._metadata = metadata
+    getattr(cls, method_name)(meta_arg)
+    return cls
+
+
+def test_initialise():
+    '''Test the _initialise method.'''
+    cls = ArgIndexToMetadataIndex
+    cls._index = 2
+    cls._info = None
+    cls._initialise(None)
+    assert cls._index == 0
+    assert isinstance(cls._info, dict)
+    assert not cls._info
+
+
+def test_scalar():
+    '''Test the _scalar method. Note, we need at least one field/operator
+    for the metadata to be valid.
+
+    '''
+    meta_arg = ScalarArgMetadata("GH_REAL", "GH_READ")
+    meta_args = [
+        FieldArgMetadata("GH_REAL", "GH_INC", "W0"),
+        meta_arg]
+    metadata = LFRicKernelMetadata(
+        operates_on="cell_column", meta_args=meta_args)
+    cls = call_method("_scalar", meta_arg, metadata)
+    assert len(cls._info) == 1
+    assert cls._info[0] == 1
+
+
+def test_field():
+    '''Test the _field method.'''
+    meta_arg = FieldArgMetadata("GH_REAL", "GH_INC", "W0")
+    metadata = LFRicKernelMetadata(
+        operates_on="cell_column", meta_args=[meta_arg])
+    cls = call_method("_field", meta_arg, metadata)
+    assert len(cls._info) == 1
+    assert cls._info[0] == 0
+
+
+def test_field_vector():
+    '''Test the _field_vector method.'''
+    meta_arg = FieldVectorArgMetadata("GH_REAL", "GH_INC", "W0", "3")
+    metadata = LFRicKernelMetadata(
+        operates_on="cell_column", meta_args=[meta_arg])
+    cls = call_method("_field_vector", meta_arg, metadata)
+    assert len(cls._info) == 3
+    assert cls._info[0] == 0
+    assert cls._info[1] == 0
+    assert cls._info[2] == 0
+
+
+def test_operator():
+    '''Test the _operator method.'''
+    meta_arg = OperatorArgMetadata("GH_REAL", "GH_WRITE", "W0", "W1")
+    metadata = LFRicKernelMetadata(
+        operates_on="cell_column", meta_args=[meta_arg])
+    cls = call_method("_operator", meta_arg, metadata)
+    assert len(cls._info) == 1
+    assert cls._info[0] == 0
+
+
+def test_cma_operator():
+    '''Test the _cma_operator method.'''
+    meta_arg = ColumnwiseOperatorArgMetadata("GH_REAL", "GH_WRITE", "W0", "W1")
+    metadata = LFRicKernelMetadata(
+        operates_on="cell_column", meta_args=[meta_arg])
+    cls = call_method("_cma_operator", meta_arg, metadata)
+    assert len(cls._info) == 1
+    assert cls._info[0] == 0
+
+
+def test_add_arg():
+    '''Test the _add_arg utility method.'''
+    meta_arg = FieldArgMetadata("GH_REAL", "GH_WRITE", "W0")
+    metadata = LFRicKernelMetadata(
+        operates_on="cell_column", meta_args=[meta_arg])
+    cls = call_method("_add_arg", meta_arg, metadata)
+    assert len(cls._info) == 1
+    assert cls._info[0] == 0
+    assert cls._index == 1
```

Comment:
Please could you extend this test by calling `add_arg` a second time (for a different type of arg) and just check that _info gets extended appropriately.

Reply:
OK.